devicetypes/smartthings/fibaro-smoke-sensor.src/fibaro-smoke-sensor.groovy

```diff
@@ -119,9 +119,28 @@ metadata {
 
 def updated() {
 	log.debug "Updated with settings: ${settings}"
+	if(!state.legacySettingsUpdated) updateLegacySettings()
 	setConfigured("false") //wait until the next time device wakeup to send configure command
 }
 
+def updateLegacySettings() {
+
+	def legacyNotificationOptionMap = {
+		"disabled" : "None",
+		"casing opened" : "Casing opened",
+		"exceeding temperature threshold" : "Exceeding temperature threshold",
+		"lack of Z-Wave range" : "Lack of Z-Wave range",
+		"all notifications" : "All"
+	}
+
+	temperatureReportInterval == "reports inactive" ? "Reports inactive" : temperatureReportInterval
+
+	zwaveNotificationStatus = zwaveNotificationStatus in legacyNotificationOptionMap ? legacyNotificationOptionMap[zwaveNotificationStatus] : zwaveNotificationStatus
```

Comment:
Have you confirmed this works? I don't recall if it's this simple or if you need to call a base class function (or if one even exists)

Reply:
Also can be simplified to `legacyNotificationOptionMap.get(zwaveNotificationStatus, zwaveNotificationStatus)`

Reply:
Although I guess this format does add the default value to the map which probably isn't ideal. `legacyNotificationOptionMap.zwaveNotificationStatus ?: zwaveNotificationStatus` might be better